### PR TITLE
Soft min contact feat

### PR DESCRIPTION
**Pull request by msultan**
- [x] Implement feature / fix bug
 - [x] Add tests
 - [x] Update changelog

Added a contact featurizer that is capable of taking a soft minimization. This follows the commit on [mdtraj #1247](https://github.com/mdtraj/mdtraj/pull/1247). I  dont know if we can merge this until we cut a new mdtraj release and pin msmbuilder to the new version.

### CODE REVIEW

**Comment by msultan** [comment]
also changed the contact featurizer dataframe to be less useless.

**Comment by mpharrigan** [comment]
for compatibility with old mdtrajs, have something like
```python
if softmax:
  md.compute_contacts(x, y, softmax=True, beta=whatever)
else:
  md.compute_contacts(x,y)
```

so it will only break if you try to use the new feature (but don't have a new mdtraj)

**Comment by msultan** [comment]
alright so i have now added exceptions to the tests until we get to the next release. The code in the transform method should then be modified at that point. This should be ready to merge after travis passes.

**Comment by cxhernandez** [comment]
You're gonna need a try-except to throw an error about using softmin with an incompatible version

**Comment by msultan** [comment]
@cxhernandez it already raises a value error if you try to setup the featurizer object without having the softmin option. I have disabled the tests for now though

**Comment by msultan** [comment]
No idea why one of the travis builds keeps stalling. Appveyor is also broken for some other weird reason.

**Comment by msultan** [comment]
Thanks a lot for looking into this @jchodera , gonna try pinning conda build to hopefully resolve the issue.

**Comment by mpharrigan** [comment]
Can you split the test into two tests (copy and paste is good). One with old behavior (that is not skipped) and one with new behavior (skipped for now, you will be in charge of enabling upon new mdtraj)

File: msmbuilder/featurizer/featurizer.py [hunk 1]
```diff
@@ -1029,12 +1030,28 @@ class ContactFeaturizer(Featurizer):
...
+    def __init__(self, contacts='all', scheme='closest-heavy', ignore_nonprotein=True,
+                 soft_min=False, soft_min_beta=20):
         self.contacts = contacts
         self.scheme = scheme
         self.ignore_nonprotein = ignore_nonprotein
+        self.soft_min = soft_min
+        self.soft_min_beta = soft_min_beta
+        if self.soft_min and not 'soft_min' in inspect.signature(md.compute_contacts).parameters:
+            raise ValueError("Sorry but soft_min requires the latest version"
+                             "of mdtraj")
```
Comment (cxhernandez) [inline msmbuilder/featurizer/featurizer.py]: forgot a space here

**Pull request merged**